### ISSUE TITLE
Clarify requirements for "Awaiting my action"

### ISSUE THREAD

**Issue by xEtherealx** [Reviewable/Reviewable#1005]
The [documentation](https://docs.reviewable.io/dashboard.html) refers to the "Awaiting my action" as "Waiting on me" and doesn't explain what criteria need to be met in order for PRs to show up there.

I was trying to educate my team on how to ensure PRs would be visible to others as needing their action, but have no guidance to give them other than "make sure you mark all comments as resolved" but I'm not confident that's all there is to it. For example, do all commenters' comments need to be resolved or if I resolve all of UserA's comments, will the PR show in just their action queue?

**Comment by pkaminski** [comment]
You're right, this is not very well documented.  The details are a bit tricky (and we tend to weak them over time, as we discover new corner cases) but I added a [high level description](https://docs.reviewable.io/reviews.html#waiting-on) to the docs and pointed to a public piece of code that matches our internal algorithm.  Hope this helps -- let me know if you think it needs further clarification.

**Comment by xEtherealx** [comment]
@pkaminski thanks, that helps but now I'm seeing something strange -- I'm reviewing a PR that is "waiting on" me
![image](https://user-images.githubusercontent.com/3417501/220725804-97cb8a6a-c91e-4555-9931-03c2047effca.png)

But in my dashboard there are no PRs "awaiting my action"
![image](https://user-images.githubusercontent.com/3417501/220725959-7280579c-d0b3-4cce-a834-7bc8c3452a76.png)

Is this perhaps a bug?

**Comment by earlAchromatic** [comment]
Hi @xEtherealx. In the screenshot you shared, there is a column header that says 'waiting on' but your row doesn't have the 'pointing hand' icon (shown in the docs). So you the review here is not waiting on you, which matches what you are seeing in the dashboard.

**Comment by xEtherealx** [comment]
Ah I see, that's a confusing UI design :)

**Comment by xEtherealx** [comment]
So actually, I'm not sure why this PR is *not* waiting on me. The author has marked all of my conversations as resolved and they're awaiting my response. Is there something in the UI that indicates what action I'm waiting on the author for?

**Comment by fahhem** [comment]
Could you post a screenshot of the whole panel? You can skip the names column if you like

**Comment by xEtherealx** [comment]
![image](https://user-images.githubusercontent.com/3417501/220732669-eb9bd19c-23fb-4c2b-bafa-fc3b2e35ceac.png)

**Comment by xEtherealx** [comment]
I'm guessing the answer might be: the author has some action to take, so it's not yet on me to reply. If that's the case, I think I'd prefer to take action sooner so that the overall PR could be unblocked faster.

**Comment by fahhem** [comment]
You can always take action on a PR that's not explicitly waiting on you. The "waiting on" column is just to make it clear who's actively blocking the PR, not who _could_ take action

**Comment by xEtherealx** [comment]
Ok that makes sense; I'm more concerned with the "awaiting my action" section of the reviews dashboard though, since that's where I go to scan for things to take action on

**Comment by xEtherealx** [comment]
I think this PR should be waiting on my though, per the docs: "All participants involved in unresolved discussions that are [unreplied](https://docs.reviewable.io/discussions.html#unreplied-discussions) for them." -- There is one unresolved discussion which the author has replied to, with the resolved disposition and my disposition is blocking.

**Comment by fahhem** [comment]
The next line in the docs explains the new state:

> However, if a user deferred by publishing a review when some files or discussions were still unreviewed or unreplied, then they'll be removed from consideration for the waiting-on list until the review's state changes. 

If you want to see all PR's you could take action on, then the "Being reviewed by me:" section in your dashboard is more relevant. "Awaiting my action" is for PR's that are blocked by you, aka waiting on your action.

**Comment by xEtherealx** [comment]
> If you want to see all PR's you could take action on, then the "Being reviewed by me:" section in your dashboard is more relevant

That section is too noisy -- what I want to know is when to re-review a PR, and I don't want to have to open up every PR that's in review by myself in order to determine that!

>However, if a user deferred by publishing a review when some files or discussions were still unreviewed or unreplied, then they'll be removed from consideration for the waiting-on list until the review's state changes.

It sounds like a key for me here is that I need to mark all files in the PR as reviewed, otherwise the PR won't be in my "awaiting my action" section? That doesn't seem like the experience I've had in the past, and also doesn't seem like a good user experience tbh.

> or discussions were still unreviewed or unreplied

Does that mean that I need to review other people's discussions as well? If so, how do I review a discussion -- does that mean marking it as read?

**Comment by fahhem** [comment]
Sorry, I'm out of my depth at this point. I'll let someone who's more familiar with the default completion condition answer your questions properly.

**Comment by xEtherealx** [comment]
@pkaminski could you chime in on this? It seems like the PR should have been "waiting on me" but was not

**Comment by pkaminski** [comment]
I'll need a pointer to the PR so that I can dump out the review state and run it through the completion condition (either your custom one or our default one).

Amusingly, we've had requests before to spell out why somebody is being waited on, but never for why somebody *isn't* waited on.  :D  I get why that'd be useful, though -- I'll need to see if we can figure out something that's not too noisy.
